src/components/annual/PriceTable.tsx

```diff
@@ -1,31 +1,12 @@
 /* eslint-disable max-len */
-import React, { useEffect, useState } from 'react';
+import React from 'react';
 import { Transition } from '@headlessui/react';
 import PriceCard from '@/components/prices/PriceCard';
-import { ProductsProps, stripeService } from '@/components/services/stripeService';
 import CardSkeleton from '@/components/components/CardSkeleton';
-import { currencyService } from '@/components/services/currencyService';
+import usePricing from '@/hooks/usePricing';
```

Comment:
cool naming, I recommend you to review Expo hooks conventions, they use them just to abstract UI/interactions logic.

Some examples https://github.com/expo/expo/blob/main/packages/expo-router/src/hooks.ts